src/store/names.ts

```diff
@@ -0,0 +1,87 @@
+import { selector } from 'recoil';
+import { getStxAddress } from '@stacks/wallet-sdk';
+
+import { fetcher } from '@common/wrapped-fetch';
+
+import { accountsStore } from '@store/wallet';
+import { currentNetworkStore, currentTransactionVersion } from '@store/networks';
+
+async function fetchNamesByAddress(networkUrl: string, address: string): Promise<string[]> {
+  const res = await fetcher(networkUrl + `/v1/addresses/stacks/${address}`);
+  const data = await res.json();
+  return data?.names || [];
+}
+
+function makeKey(networkUrl: string, address: string): string {
+  return `${networkUrl}__${address}`;
+}
+
+function getLocalNames(networkUrl: string, address: string): string[] | null {
```

Comment:
Any reason not to return empty array rather than null?

Reply:
To give indication that it needs to fetch